File: java/src/main/java/de/sven_torben/serialization_benchmark/serializer/ProtocolBuffersSerializer.java
package de.sven_torben.serialization_benchmark.serializer;

import java.io.InputStream;
import java.io.OutputStream;

import de.sven_torben.serialization_benchmark.testdata.protobuf.CatalogProtos;
import de.sven_torben.serialization_benchmark.testdata.protobuf.CatalogProtos.Catalog;

public final class ProtocolBuffersSerializer extends AbstractSerializer<Catalog> {

	public ProtocolBuffersSerializer() {
		super(CatalogProtos.Catalog.class);
	}

	@Override
	public void serialize(final OutputStream stream, final CatalogProtos.Catalog data) throws Exception {
		data.writeTo(stream);
	}

	@Override
	public CatalogProtos.Catalog deserialize(final InputStream stream) throws Exception {
		return CatalogProtos.Catalog.parseFrom(stream);
	}

}

File: net/Serializer/ProtocolBuffersSerializer.cs
using De.Sven_Torben.Serialization_Benchmark.Testdata;
using ProtoBuf;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace De.Sven_Torben.Serialization_Benchmark.Serializer
{
    class ProtocolBuffersSerializer : AbstractSerializer<Catalog>
    {

        internal ProtocolBuffersSerializer()
            : base(new Catalog()) { }

        public override void Serialize(Stream stream, Catalog data)
        {
            ProtoBuf.Serializer.Serialize(stream, data);
        }

        public override Catalog Deserialize(Stream stream)
        {
            return ProtoBuf.Serializer.Deserialize<Catalog>(stream);
        }
    }
}
